src/compressor/zlib/CompressionZlib.cc

```diff
@@ -134,7 +134,7 @@ int CompressionZlib::decompress(const bufferlist &in, bufferlist &out)
       strm.avail_out = max_len;
       strm.next_out = c_out;
       ret = inflate(&strm, Z_NO_FLUSH);
-      if (ret != Z_OK && ret != Z_STREAM_END) {
```

Comment:
@yuyuyu101 In this case strm.avail_out == max_len and we exit from while by the condition, nothing will be added to output, because have == 0 (https://github.com/Ved-vampir/ceph/blob/zlib_fix/src/compressor/zlib/CompressionZlib.cc#L143)
In zlib docs: "But now deflate() **produces no output at all**, and avail_out remains unchanged as CHUNK. That deflate() call wasn't able to do anything, either consume input or produce output, and so it returns Z_BUF_ERROR" (in inflate block link on this description). Also in example they don't check for this return value at all.